docs/extending/developing-code/extending/case-studies/KroneckerProduct/completing.rst

```diff
@@ -0,0 +1,141 @@
+``Completing KroneckerProduct``
+===============================
+
+.. contents::
+   :depth: 1
+   :local:
+
+
+``KroneckerProduct[]`` skeletal code
+------------------------------------
+
+Now we fill in the parts of ``KroneckerProduct`` other than the class
+definition line and its docstring.
+
+None of the other built-in Mathics Functions serve as good model to
+copy from, so we'll use a combination ``Transpose`` and
+``PauliMatrix``. The former function is selected because it takes a
+Matrix parameter, and PauliMatrix function because it is a SymPy
+function which was recently created at the time of this writing.
+
+Here is the first cut for ``KroneckerProduct``:
+
+.. code-block::
+
+    class KroneckerProduct(SympyFunction):
+        """
+	... as before
+        """
+
+        attributes = A_PROTECTED | A_READ_PROTECTED
+        summary_text = "Kronecker product"
+        sympy_name = "physics.quantum.TensorProduct"
+
+        def apply(self, mi: ListExpression, evaluation: Evaluation):
+            "KroneckerProduct[mi__]"
+            sympy_mi = [to_sympy_matrix(m) for m in mi.elements]
+            return from_sympy(TensorProduct(*sympy_mi))
+
+In the above, I needed to add imports for ``SympyFunction``,
+``ListExpression``, ``Evaluation`` and other places. Examples of these
+of of course can be adapted from the place where you copied the code
+from.
+
+
+Now let's go over the new parts...
+
+``KroneckerProduct[]`` class variables
+--------------------------------------
+
+As in ``Undefined``, we set the attributes of function
+``KroneckerProduct`` by setting the class variable ``attributes``.
+
+Here, in addition to being "Protected", it is also "ReadProtected" and
+the way we indicate both attributes is to bit-or, ``|``, the to attribute
+values ``A_PROTECTED`` and ``A_READ_PROTECTED``.
+
+The purpose of the class value ``summary_text`` is described when we
```

Comment:
Maybe mention that if `summary_text` is not set in the class, then there is a pytest that is going to complain...

Reply:
Ok - will do.